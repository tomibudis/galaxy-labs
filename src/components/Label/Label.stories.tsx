import React from 'react';

import { Meta, ComponentStory } from '@storybook/react';

import Label from './Label';

export default {
  title: 'Components/Label',
  component: Label,
  argTypes: {
    children: {
      table: {
        disable: true,
      },
    },
  },
} as Meta<typeof Label>;

const Template: ComponentStory<typeof Label> = (args): JSX.Element => {
  return <Label {...args}>{args.children}</Label>;
};

export const BasicLabel = Template.bind({});
BasicLabel.args = {
  children: 'Label Component',
};

export const ColorLabel = Template.bind({});
ColorLabel.args = {
  children: 'Label Color',
  className: 'text-warning',
};

export const WithAsterisk = Template.bind({});
WithAsterisk.args = {
  children: 'Label with asterisk',
  isRequired: true,
};
