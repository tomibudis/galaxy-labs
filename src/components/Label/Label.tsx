import React from 'react';

import PropTypes from 'prop-types';
import cx from 'classnames';

import { LabelProps } from '#interface/index';
import styles from './styles.module.scss';

const Label: React.FC<LabelProps> = ({
  className,
  children,
  htmlFor,
  id,
  isRequired,
}): JSX.Element => {
  return (
    <label id={id} htmlFor={htmlFor} className={cx(styles.label, className)}>
      {children} {isRequired && <span className="text-danger">*</span>}
    </label>
  );
};

Label.propTypes = {
  /** Additional classname to define color of label */
  className: PropTypes.string,
  /** Children ReactNode */
  children: PropTypes.node.isRequired,
};

export default Label;
