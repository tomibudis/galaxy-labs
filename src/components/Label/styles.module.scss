.label {
  font-size: x-small;
}
