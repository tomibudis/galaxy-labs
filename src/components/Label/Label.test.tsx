import React from 'react';
import { cleanup, render } from '@testing-library/react';

import Label from './Label';

afterEach(() => {
  cleanup();
});

const LabelComponent = (props): JSX.Element => {
  return <Label {...props}>{props.children}</Label>;
};

describe('<Label />', () => {
  it('render component without crashing', () => {
    render(<LabelComponent />);
    expect(true).toBeTruthy();
  });
  it('match with snapshot file', () => {
    const { baseElement } = render(
      <LabelComponent className="text-warning">Label</LabelComponent>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
