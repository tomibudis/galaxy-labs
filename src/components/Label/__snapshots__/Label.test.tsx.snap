// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> match with snapshot file 1`] = `
<body>
  <div>
    <label
      class="label text-warning"
    >
      Label
       
    </label>
  </div>
</body>
`;
